tests/unit/i18n-t-test.js

```diff
@@ -71,3 +88,27 @@ test("check unknown locale", function(assert) {
   const result = this.subject({ locale: 'uy' }).t('not.yet.translated', {count: 2});
   assert.equal('Missing translation: not.yet.translated', result);
 });
+
+test("locale can be overridden and warns by default", function(assert) {
+  const i18n = this.subject({ locale: 'en' });
+
+  const result = i18n.t('no.interpolations', { locale: 'es' });
+  assert.equal(result.string, 'texto sin interpolaciones');
+  assert.equal(warnings.length, 1);
+});
+
+test("locale can be overridden and does not warn if allowLocaleOverride is true", function(assert) {
+  const i18n = this.subject({ locale: 'en', allowLocaleOverride: true });
+
+  const result = i18n.t('no.interpolations', { locale: 'es' });
+  assert.equal(result.string, 'texto sin interpolaciones');
+  assert.equal(warnings.length, 0);
+});
+
+test("locale can be used as an interpolation key if allowLocaleOverride is false", function(assert) {
+  const i18n = this.subject({ locale: 'en', allowLocaleOverride: false });
+
+  const result = i18n.t('with.locale.interpolation', { locale: 'es' });
+  assert.equal(result.string, 'Locale: es');
```

Comment:
For some reason, in these tests alone, the `t` function seems to return an object with a single property called `'string'`, which I have to access to get the actual translation. Really not sure why that is.